src/app/pages/data/edit-record/edit-record.component.ts

```diff
@@ -42,75 +47,96 @@ export class EditRecordComponent implements OnInit {
-                    this.apiService.getRecordMedia(recordId).subscribe(
-                        (media: RecordMedia[]) => this.imagesMetadata = media.map((image: RecordMedia) => ({
+        let recordObservable: Observable<Record>;
+        if (params.hasOwnProperty('recordId')) {
+            // fetching the record will also fetch its attached media and child dataset (if there are children)
+            recordObservable = this.apiService.getRecordById(+params['recordId']).pipe(
+                // get record media
+                mergeMap((record: Record) =>
+                    this.apiService.getRecordMedia(record.id).pipe(
+                        tap((media: RecordMedia[]) => this.imagesMetadata = media.map((image: RecordMedia) => ({
                             source: image.file,
                             alt: image.file.substring(image.file.lastIndexOf('/') + 1),
                             title: image.file.substring(image.file.lastIndexOf('/') + 1)
-                        }))
-                    );
-                } else {
+                            }))
+                        ),
+                        // with resultSelect deprecated, need to return record via map
+                        map((media: RecordMedia[]) => record)
+                    )
+                ),
+                // get record children
+                mergeMap((record: Record) => {
+                    if (record.children && record.children.length) {
+                        return this.apiService.getRecordById(record.children[0]).pipe(
+                            mergeMap((childRecord: Record) => this.apiService.getDatasetById(childRecord.dataset)),
+                            tap((childDataset: Dataset) => this.childDataset = childDataset),
+                            // with resultSelect deprecated, need to return record via map
+                            map((childDataset: Dataset) => record)
+                        );
+                    } else {
+                        return of(record);
+                    }
+                })
+            );
+        } else {
+            recordObservable = of(<Record>{
+                dataset: +params['datasetId']
+            });
+        }
+
+        forkJoin(projectObservable, datasetObservable, recordObservable).subscribe(
+            (result: [Project, Dataset, Record]) => {
+                const project = result[0];
+                const dataset = result[1];
+                const record = result[2];
+
+                this.breadcrumbItems.push({
+                    label: project.name,
+                    routerLink: [`/data/projects/${project.id}/datasets`]
+                }, {
+                    label: dataset.name,
+                    routerLink: [`/data/projects/${project.id}/datasets/${dataset.id}`]
+                },
+                    {label: record.id ? 'Edit Record' : 'Create Record'}
+                );
+
+                if (record.id && dataset.type !== 'generic') {
```

Comment:
like for the server side we should avoid dataset type testing.
maybe test that the extent is not `None` or `empty`